### ISSUE TITLE
Pasting Unicode text sometimes doesn't work on Windows

### ISSUE THREAD

**Issue by entuland** [minetest/minetest#11300]
Minetest version 5.4.1, seeing a few oddities trying to deal with UTF-8 characters.

I set all the game fonts (from the settings) to use Unifont and the fallback one to use Unifont Upper, for starters.

Printing Greek or Russian characters in the chat via a mod works, but trying to type such chars directly in the chat, just like pasting such text, results in question marks.

Same behavior (question marks) trying to type Greek characters in a textbox.

(By the way, I enter such Greek characters via a AutoHotKey script, which seems to work fine everywhere else - just like here, typing some random gibberish: καjηδφκασhδφκθασδφ)

Typing Italian accented vowels in a textbox they show up correctly, but as I copy them from the same textbox and I paste it back it comes up garbled. (my real keyboard has the Italian layout)

Attaching a few screenshots showing the behavior (where I tried to print Russian characters I also appended a Unicode smiley, which the chat didn't print out)

Mod-printed Russian chars, the mod printed these chars + a Unicode Smiley (this one: 😀) but the chat didn't print it out:
![2021-05-31 00_48_08-Window](https://user-images.githubusercontent.com/4322309/120123112-4c632100-c1ad-11eb-98cf-cb95f01df4ea.png)

Mod-printed Greek chars + Greek chars typed in the chat (showing up as question marks:)
![2021-05-31 01_03_42-Window](https://user-images.githubusercontent.com/4322309/120123108-4b31f400-c1ad-11eb-8e63-6a8c24519a93.png)

Italian accented letters entered properly with my Italian keyboard:
![2021-05-31 00_43_48-Window](https://user-images.githubusercontent.com/4322309/120123110-4bca8a80-c1ad-11eb-9b2e-f76856ba3056.png)

Result of copypasting the above Italian accented letters:
![2021-05-31 00_44_03-Window](https://user-images.githubusercontent.com/4322309/120123111-4bca8a80-c1ad-11eb-985c-9db94e8c4109.png)

Edit: adding the AutoHotKey script I use to enter Greek chars with CapsLock on:
[Unicode.ahk.txt](https://github.com/minetest/minetest/files/6567053/Unicode.ahk.txt)

**Comment by entuland** [comment]
Downloaded the latest 5.5.0-dev windows 64 build and reproed the above, here are the results:

For starters, set all the fonts to use Unifont and the Upper variant as fallback:

![2021-05-31 10_28_16-Window](https://user-images.githubusercontent.com/4322309/120167071-98d64d00-c1fd-11eb-823a-ec631505e45f.png)

Copypasting Italian accented letters in a textbox works, but typing or pasting greek or cyrillic still fails with question marks:

![2021-05-31 10_32_00-Window](https://user-images.githubusercontent.com/4322309/120167125-a8559600-c1fd-11eb-8ee3-e90e8769ce61.png)

![2021-05-31 10_50_11-Window](https://user-images.githubusercontent.com/4322309/120167451-0da98700-c1fe-11eb-80be-9a0a9953e6e3.png)

Printing greek or cyrillic via a mod works, printing smileys still fails (the last line should have been `smileys: 😀 🤪 😠`)

In the following I pasted in the chat prompt all of this text at once: `οψρψρξεyωρψ вротмввсмшнкыардж 😀 🤪 😠`:

![2021-05-31 10_45_19-Window](https://user-images.githubusercontent.com/4322309/120167522-20bc5700-c1fe-11eb-8d71-187ceee54985.png)

```lua
minetest.register_chatcommand("utf8test", {
	description = "utf8 test description",
	func = function()
		minetest.chat_send_all("greek: οψρψρξεyωρψ")
		minetest.chat_send_all("cyrillic: вротмввсмшнкыардж")
		minetest.chat_send_all("smileys: 😀 🤪 😠")
	end
})
```

**Comment by sfan5** [comment]
There's two different things here:
* Unicode pasting being broken: this would be responsibility of IrrlichtMt
* Emojis not appearing: Emojis will render (albeit monochrome, color is not supported) if the specified font supports them

**Comment by entuland** [comment]
Also directly entering some of those Unicode chars seems to be broken, so the issues may be three in total - at least, I assume that if my AutoHotKey script is working properly anywhere else, it should be sending the correct chars to Minetest as well, and as such it would be different from pasting.

Should I close / edit this issue and create new ones to split them all?

**Comment by sfan5** [comment]
works for me *or* fixed by minetest/irrlicht@34f1f8d089ab1f7958c874265ff37cced54d18ed
